evalml/pipelines/components/transformers/preprocessing/decomposer.py

```diff
@@ -26,10 +32,126 @@ def __init__(self, parameters=None, component_obj=None, random_seed=0, **kwargs)
+                dt_col = pd.Series(dt_df.index)
+            elif num_datetime_features == 0:
+                raise ValueError(
+                    "There are no Datetime features in the feature data and neither the feature nor the target data have a DateTime index.",
+                )
+            # Use a datetime column of the features if there's only one
+            elif num_datetime_features == 1:
+                dt_col = dt_df.ww.select("Datetime").squeeze()
+            # With more than one datetime column, use the time_index parameter, if provided.
+            elif num_datetime_features > 1:
+                if self.parameters.get("time_index", None) is None:
+                    raise ValueError(
+                        "Too many Datetime features provided in data but no time_index column specified during __init__.",
+                    )
+                elif not self.parameters["time_index"] in X:
+                    time_index_col = self.parameters["time_index"]
+                    raise ValueError(
+                        f"Too many Datetime features provided in data and provided time_index column {time_index_col} not present in data.",
+                    )
+
+        time_index = pd.DatetimeIndex(dt_col, freq=pd.infer_freq(dt_col)).rename(
+            y.index.name,
+        )
+        return y.set_axis(time_index)
+
     @abstractmethod
     def get_trend_dataframe(self, y):
         """Return a list of dataframes, each with 3 columns: trend, seasonality, residual."""
 
     @abstractmethod
     def inverse_transform(self, y):
         """Add the trend + seasonality back to y."""
+
+    def determine_periodicity(self, X, y, method="autocorrelation"):
+        """Function that uses autocorrelative methods to determine the first, signficant period of the seasonal signal.
+
+        Args:
+            X (pandas.DataFrame): The feature data of the time series problem.
+            y (pandas.Series): The target data of a time series problem.
+            method (str): Either "autocorrelation" or "partial-autocorrelation".  The method by which to determine the
+                first period of the seasonal part of the target signal.  "partial-autocorrelation" should currently not
+                be used.  Defaults to "autocorrelation".
+
+        Returns:
+            (list[int]): The integer numbers of entries in time series data over which the seasonal part of the target data
+                repeats.  If the time series data is in days, then this is the number of days that it takes the target's
+                seasonal signal to repeat. Note: the target data can contain multiple seasonal signals.  This function
+                will only return the first, and thus, shortest period.  E.g. if the target has both weekly and yearly
+                seasonality, the function will only return "7" and not return "365".  If no period is detected, returns [None].
+
+        """
+
+        def _get_rel_max_from_acf(y):
```

Comment:
While these helper functions are completely private, it would be helpful if they had one-line docstrings for posterity purposes.